File: viewbindingdemo/src/main/java/com/example/viewbindingdemo/MainActivity.java
package com.example.viewbindingdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.viewbindingdemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());//获取根节点

        binding.textView.setText("view1");
        binding.textView2.setText("view2");
    }
}

File: viewbindingdemo2kotlin/src/main/java/com/example/viewbindingdemo2kotlin/MainActivity.kt
package com.example.viewbindingdemo2kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.viewbindingdemo2kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.textView.text = "hello"
        binding.textView2.text = "binding"
    }
}
